File: 백준/Bronze/24416. 알고리즘 수업 － 피보나치 수 1/알고리즘 수업 － 피보나치 수 1.py
def fib(n):
    fibo = [1,1]
    for i in range(2,n+1):
        fibo.append(fibo[i-1] + fibo[i-2])
    return fibo[n-1]

def fibonacci(n):
    global f
    f[1] = 1
    f[2] = 1
    count2 = 0
    for i in range(3, n+1):
        f[i] = f[i-1] + f[i-2]
        count2 += 1
    return count2

n = int(input())
f = [0]*(n+1)

print(fib(n),fibonacci(n))

File: 백준/Bronze/24416. 알고리즘 수업 － 피보나치 수 1/알고리즘 수업 － 피보나치 수 1.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Main {
    static int count1 = 0;
    static int count2 = 0;
    public static int fib(int n){
        if(n == 1 || n == 2){
            count1 ++;
            return 1;
        } else{
            return (fib(n-1)+fib(n-2));
        }
    }
    public static int[] fibonacci(int[] arr){
        arr[1] = 1;
        arr[2] = 1;
        for(int i = 3; i< arr.length;i++){
            count2++;
            arr[i] = arr[i-1] + arr[i-2];
        }
        return arr;
    }
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int num = Integer.parseInt(br.readLine());
        int[] arr = new int[num+1];
        fib(num);
        fibonacci(arr);
        System.out.println(count1 + " " + count2);
    }
}
